site/app/controllers/admin/PlagiarismController.php

```diff
@@ -196,100 +213,126 @@ private function saveNewProvidedCode(string $temporary_file_path, string $filena
+        // TODO: return to this and enable later
+        // $nightly_rerun_info_file = "/var/local/submitty/courses/" . $semester . "/" . $course . "/lichen/nightly_rerun.json";
+        // if (!file_exists($nightly_rerun_info_file)) {
+        //     $nightly_rerun_info = [];
+        //     foreach ($gradeables_with_plagiarism_result as $gradeable_id_title) {
+        //         $nightly_rerun_info[$gradeable_id_title['g_id']] = false;
+        //     }
+        //     if (file_put_contents($nightly_rerun_info_file, json_encode($nightly_rerun_info, JSON_PRETTY_PRINT)) === false) {
+        //         die("Failed to create nightly rerun info file");
+        //     }
+        // }
+        // else {
+        //     $nightly_rerun_info = json_decode(file_get_contents($nightly_rerun_info_file), true);
+        //     foreach ($nightly_rerun_info as $gradeable_id => $nightly_rerun_status) {
+        //         $flag = 0;
+        //         foreach ($gradeables_with_plagiarism_result as $gradeable_id_title) {
+        //             if ($gradeable_id_title['g_id'] == $gradeable_id) {
+        //                 $flag = 1;
+        //                 break;
+        //             }
+        //         }
+        //         if ($flag == 0) {
+        //             #implies plagiarism result for this gradeable are deleted
+        //             unset($nightly_rerun_info[$gradeable_id]);
+        //         }
+        //     }
+        //
+        //     foreach ($gradeables_with_plagiarism_result as $gradeable_id_title) {
+        //         if (!array_key_exists($gradeable_id_title['g_id'], $nightly_rerun_info)) {
+        //             #implies plagiarism was run for this gradeable
+        //             $nightly_rerun_info[$gradeable_id_title['g_id']] = false;
+        //         }
+        //     }
+        //     if (file_put_contents($nightly_rerun_info_file, json_encode($nightly_rerun_info, JSON_PRETTY_PRINT)) === false) {
+        //         die("Failed to create nightly rerun info file");
+        //     }
+        // }
+        $nightly_rerun_info = []; // placeholder
+
+        return new WebResponse(
+            ['admin', 'Plagiarism'],
+            'plagiarismMainPage',
+            $all_configurations,
+            $refresh_page,
+            $nightly_rerun_info
+        );
     }
 
     /**
+     * @param string $gradeable_id
+     * @param string $config_id
+     * @return WebResponse|RedirectResponse
      * @Route("/courses/{_semester}/{_course}/plagiarism/gradeable/{gradeable_id}")
      */
-    public function showPlagiarismResult($gradeable_id) {
-        $semester = $this->core->getConfig()->getSemester();
-        $course = $this->core->getConfig()->getCourse();
+    public function showPlagiarismResult(string $gradeable_id, string $config_id) {
```

Comment:
I saw you previously returning MultiResponse and now nothing. Instead you would want this since both Web and Redirect responses implement this as shown below. You may need to add this to the top of the file in a "use" statement.
```suggestion
    public function showPlagiarismResult(string $gradeable_id, string $config_id): ResponseInterface {
```